src/linuxdeploy.cpp

```diff
@@ -0,0 +1,78 @@
+#include <iostream>
+#include <boost/filesystem/path.hpp>
+
+#include <linuxdeploy/core/appdir.h>
+#include <linuxdeploy/core/log.h>
+#include <linuxdeploy/util/util.h>
+
+#include "linuxdeploy.h"
+
+using namespace linuxdeploy::core;
+
+using namespace linuxdeploy::core::log;
+using namespace linuxdeploy::util;
+namespace bf = boost::filesystem;
+
+namespace linuxdeploy {
+    desktopfile::DesktopFile getMainDesktopFile(std::vector<std::string>& desktopFilePaths,
+                                                std::vector<desktopfile::DesktopFile>& deployedDesktopFiles);
+
+    bool deployAppDirRootFiles(std::vector<std::string> desktopFilePaths,
+                               std::string customAppRunPath, appdir::AppDir& appDir) {
+        // search for desktop file and deploy it to AppDir root
+        ldLog() << std::endl << "-- Deploying files into AppDir root directory --" << std::endl;
+
+        auto deployedDesktopFiles = appDir.deployedDesktopFiles();
+
+        try {
+            desktopfile::DesktopFile desktopFile = getMainDesktopFile(desktopFilePaths, deployedDesktopFiles);
+            ldLog() << "Deploying desktop file:" << desktopFile.path() << std::endl;
+            if (!appDir.createLinksInAppDirRoot(desktopFile, customAppRunPath))
+                return false;
+
+        } catch (const std::runtime_error& er) {
+            return false;
+        }
+
+        return true;
+    }
+
+    desktopfile::DesktopFile getMainDesktopFile(std::vector<std::string>& desktopFilePaths,
+                                                std::vector<desktopfile::DesktopFile>& deployedDesktopFiles) {
```

Comment:
Can't these params be `const` references?

Reply:
Yes they can, changed.